autonick.lua

```diff
@@ -0,0 +1,57 @@
+-- automatically nickname dwarves
+--[====[
+
+autonick
+========
+Automatically nickname dwarves according to ``data/init/names.txt``.
+
+One nickname per line, or an empty line. Repeat entries are discarded.
+
+Dwarves with manually set nicknames are ignored.
+
+If there are fewer available nicknames than dwarves, the remaining
+dwarves will go un-nicknamed.
+
+You may wish to use this script with the "repeat" command, e.g:
+``repeat -name autonick -time 3 -timeUnits months -command [ autonick ]``
+
+]====]
+
+names = {};
+index = 1;
+
+path = dfhack.getDFPath () .. "/data/init/names.txt";
+
+-- grab list, put in array
+for line in io.lines(path) do
+	if line ~= "" then
```

Comment:
line = line:trim() would be useful here, to remove leading and trailing whitespace